### DISCUSSION TITLE
Whatsapp provider

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [nextauthjs/next-auth#11234] [Ideas]
### Goals

It'd work similar to Facebook, Twitter login, but for Whatsapp.

### Background

This is really needed by our customers.

### Proposal

N/A

**Comment by shawnmclean** [comment]
Are you looking to send a one time password to whatsapp?

If so, this product, based on the model, is already bound to emails as an identity.

Phone number auth may be tricky if anyone else has figured this out, I'd love to see it.

**Comment by atumas-bananamilk** [comment]
@shawnmclean

Since Facebook owns WhatsApp, we managed to gain access / refresh tokens that also interact with WhatsApp API using these scopes when using `FacebookProvider`:
- `whatsapp_business_management`
- `whatsapp_business_messaging` 

You also need to add `WhatsApp` as a product into your Facebook app in Facebook Developers.

It's identical to how Instagram is handled (also owned by Facebook):
- `instagram_basic`
- `instagram_content_publish`

Regarding connecting WhatsApp without Facebook login - don't know yet, probably impossible.
